pkg/datapath/loader/loader.go

```diff
@@ -54,34 +48,42 @@ type endpoint interface {
 	StringID() string
 }
 
-// joinEP shells out to bpf/join_ep.sh to compile/load the BPF datapath for the
-// specified endpoint, and handles timeouts if the loading takes too long.
-func joinEP(ep endpoint, libdir, rundir, epdir string) error {
-	args := []string{libdir, rundir, epdir, ep.InterfaceName(), ep.StringID()}
-	prog := filepath.Join(libdir, "join_ep.sh")
-
-	scopedLog := ep.Logger()
+// compileDatpath invokes the compiler and linker to create all state files for
+// the BPF datapath, with the primary target being the BPF ELF binary.
+//
+// If debug is enabled, create also the following output files:
+// * Preprocessed C
+// * Assembly
+// * Object compiled with debug symbols
+func compileDatapath(ep endpoint, dirs *directoryInfo, debug bool) error {
+	// TODO: Consider logging kernel/clang versions here too
```

Comment:
We have them when we start cilium-agent.

Reply:
Yep, would just need a little refactoring to make available. Basically this is just a comment saying that when we ran `join_ep.sh` before, we would always print these first, inline, in the logs. We also do this on agent startup, but it's possible to lose the logs from when the daemon first started.